From 65e09b83664957a28e08e3f659cc67bad1acba5a Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Tue, 26 May 2026 17:46:53 +0800
Subject: [PATCH 2/2] lldp: deprecate link aggregation TLV from 802.3

Link Aggregation TLV has been deprecated from  IEEE Std 802.3 and is now
standardized as IEEE Std 802.1AX, which is part of the IEEE 802.1. This
patch update the type and subtype of link aggregation TLV.

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Upstream-Status: Pending
---
 src/daemon/lldp-tlv.h       | 1 +
 src/daemon/protocols/lldp.c | 8 ++++----
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/daemon/lldp-tlv.h b/src/daemon/lldp-tlv.h
index 07abd4f..311540a 100644
--- a/src/daemon/lldp-tlv.h
+++ b/src/daemon/lldp-tlv.h
@@ -62,6 +62,7 @@
 #define LLDP_TLV_DOT1_PPVID 2
 #define LLDP_TLV_DOT1_VLANNAME 3
 #define LLDP_TLV_DOT1_PI 4
+#define LLDP_TLV_DOT1_LA 7
 
 #define LLDP_TLV_DOT3_MAC 1
 #define LLDP_TLV_DOT3_POWER 2
diff --git a/src/daemon/protocols/lldp.c b/src/daemon/protocols/lldp.c
index 5939ca7..4c918c9 100644
--- a/src/daemon/protocols/lldp.c
+++ b/src/daemon/protocols/lldp.c
@@ -250,17 +250,17 @@ _lldp_send(struct lldpd *global, struct lldpd_hardware *hardware, u_int8_t c_id_
 			POKE_BYTES(pi->p_pi, pi->p_pi_len) && POKE_END_LLDP_TLV))
 			goto toobig;
 	}
-#endif
 
-#ifdef ENABLE_DOT3
 	/* Aggregation status */
-	if (!(POKE_START_LLDP_TLV(LLDP_TLV_ORG) && POKE_BYTES(dot3, sizeof(dot3)) &&
-		POKE_UINT8(LLDP_TLV_DOT3_LA) &&
+	if (!(POKE_START_LLDP_TLV(LLDP_TLV_ORG) && POKE_BYTES(dot1, sizeof(dot1)) &&
+		POKE_UINT8(LLDP_TLV_DOT1_LA) &&
 		/* Bit 0 = capability ; Bit 1 = status */
 		POKE_UINT8((port->p_aggregid) ? 3 : 1) &&
 		POKE_UINT32(port->p_aggregid) && POKE_END_LLDP_TLV))
 		goto toobig;
+#endif
 
+#ifdef ENABLE_DOT3
 	macmerge_tlv = port->p_macmerge.support | port->p_macmerge.status << 1 |
 		 port->p_macmerge.active << 2 | port->p_macmerge.frag_size << 3;
 	if (!(
-- 
2.43.0

